backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala

```diff
@@ -715,6 +716,21 @@ trait StandardAsyncExecutionActor
     case _ => 0
   }
 
+  /**
+   * Returns the memory multiplier for previous attempt if available
+   */
+  lazy val previousMemoryMultiplier: Option[MemoryRetryMultiplierRefined] = jobDescriptor.prefetchedKvStoreEntries.get(BackendLifecycleActorFactory.MemoryMultiplierKey) match {
+    case Some(KvPair(_,v)) => refineV[MemoryRetryMultiplier](v.toDouble) match {
+      case Left(e) =>
+        // should not happen as Cromwell itself had written the value as a Double
+        log.error(e, s"Programmer error: unexpected failure attempting to read value for MemoryMultiplierKey from Key Value table. " +
+          "Expected value should be in range 1.0 ≤ n ≤ 99.0")
```

Comment:
Hmm... the multiplication factor must be between 1 and 99, but the current multiplier might be higher by the time it's been applied a couple of times?

Eg if my retry factor is 2 and I retry 7 times, the multiplier would be 128 - but that's not invalid (as far as I understand things?)

Reply:
The current validation only validates that the `memoryMultiplier` (indicated by `memoryRetryFactor` in code) that is passed through workflow options is between range 1.0 <= n <= 99.0. We currently don't validate the value of currentMemoryMultiplier. So if the retry factor is 2 and task is retried 7 times, the currentMemoryMultiplier will be 128.0 and it will be valid.

Reply:
I have also updated the error message. Having `Expected value should be in range 1.0 ≤ n ≤ 99.0` here was a copy/paste error 🤦‍♀️

Reply:
Having said that 128.0 will be a valid currentMemoryMultiplier, I am thinking should it be a valid multiplier though? 🤔

Reply:
Update: "if the retry factor is 2 and task is retried 7 times, the currentMemoryMultiplier will be 128.0 and it will be valid" was not true when I opened the PR. This is because whenever Cromwell would instantiate `previousMemoryMultiplier` or `nextMultiplier` (in JobPreparationActor) it would try to validate and refine it to `MemoryRetryMultiplierRefined` type which only accepts values between 1 to 99. Hence once the value reached above 99, it would be set back to 1 as the conversion would fail.

After discussion in standup, the consensus was that we should allow values >99.0 as the multiplier. This is because practically no user should have retry factor more than 2.0. Hence I have updated the code to only validate the retry factor (that is passed through workflow options) to be within 1.0 <= x <= 99.0 range. And the `previousMemoryMultiplier` or `nextMultiplier` will simply be a Double.